### PR TITLE
Feature/interlok 4453 update change char encoding service expressions

### PR DESCRIPTION
**Pull request by saurav-mukherjee-telus**
## Motivation

Why am I doing this

## Modification

What did I change

## PR Checklist

- [x] been self-reviewed.
- [ ] Added javadocs for most classes and all non-trivial methods
- [ ] Added supporting annotations (like @XStreamAlias / @ComponentProfile)
- [ ] Added DefaultValue annotation when there is a default value (e.g. @DefaultValue('true'))
- [ ] Added validation annotation (@NotNull...) when required and add @Valid when nested class has some validation
- [ ] Checked that @NotNull and @NotBlank annotations have a meaningful message when appropriate.
- [ ] Checked that new 3rd party dependencies are appropriately licensed
- [ ] Added comments explaining the "why" and the intent of the code wherever it would not be obvious for an unfamiliar reader
- [ ] Added unit tests or modified existing tests to cover new code paths
- [ ] Tested new/updated components in the UI and at runtime in an Interlok instance
- [ ] Reviewed java class members so that missing annotations are added (InputFieldDefault/ComponentProfile etc)
- [ ] Checked that javadoc generation doesn't report errors
- [ ] Checked the display of the component in the UI
- [ ] Remove any config/license annotations
- [ ] Check the gradle build file to make sure the dependencies section is more explicit "implementation/api".

## Result

What's the end result for the user

## Testing

How can I test this if I'm reviewing this.

### CODE REVIEW

File: interlok-core/src/test/java/com/adaptris/core/services/ChangeCharEncodingServiceTest.java [hunk 1]
```diff
@@ -55,4 +55,19 @@ public void testChangeCharset() throws Exception {
     assertEquals(Charset.forName("iso-8859-1"), Charset.forName(msg.getContentEncoding()));
   }
 
+  @Test
+  public void testSetCharEncodingExpression() throws Exception {
+    ChangeCharEncodingService srv = new ChangeCharEncodingService();
+    assertNull(srv.getCharEncoding());
+    srv.setCharEncoding("%message{pn.processing.worker.transform.encoding-target}");
```
Comment (sebastien-belin-adp) [inline interlok-core/src/test/java/com/adaptris/core/services/ChangeCharEncodingServiceTest.java]: You don't/shouldn't need to use pn metadata here as Interlok is not PN.
But it doesn't matter.

**Review by sebastien-belin-adp — APPROVED** [review]

**Review by sebastien-belin-adp — CHANGES_REQUESTED** [review]
If you want to improve the coverage you could add another test with an empty char encoding value. That will test one more branch of the code.

File: interlok-core/src/test/java/com/adaptris/core/services/ChangeCharEncodingServiceTest.java (hunk 1 above)
Comment (saurav-mukherjee-telus) [reply interlok-core/src/test/java/com/adaptris/core/services/ChangeCharEncodingServiceTest.java]: Thanks @sebastien-belin-adp

**Comment by saurav-mukherjee-telus** [comment]
> If you want to improve the coverage you could add another test with an empty char encoding value. That will test one more branch of the code.

Hi @sebastien-belin-adp  - Null character value test is in place now

**Review by sebastien-belin-adp — APPROVED** [review]
Thanks
